test/e2e/copy_from_bundle_image_test.go

```diff
@@ -0,0 +1,213 @@
+// Copyright 2020 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package e2e
+
+import (
+	"bytes"
+	"fmt"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/k14s/imgpkg/pkg/imgpkg/lockconfig"
+)
+
+func TestCopyBundleWithCollocatedReferencedImagesToRepoDestinationAndOutputBundleLockFile(t *testing.T) {
+	env := BuildEnv(t)
+	imgpkg := Imgpkg{t, Logger{}, env.ImgpkgPath}
+	defer env.Cleanup()
+
+	imageDigest := env.ImageFactory.pushSimpleAppImageWithRandomFile(imgpkg, env.Image)
+
+	// create a bundle with ref to generic
+	imageLockYAML := fmt.Sprintf(`---
+apiVersion: imgpkg.carvel.dev/v1alpha1
+kind: ImagesLock
+images:
+- image: %s%s
+`, env.Image, imageDigest)
+	bundleDir := env.BundleFactory.createBundleDir(bundleYAML, imageLockYAML)
+
+	// create bundle that refs image and a random tag based on time
```

Comment:
what do you think of switching from comments to https://github.com/vmware-tanzu/carvel-kapp/blob/develop/test/e2e/create_update_delete_test.go#L67 for separation? (possibly as a follow up PR)

Reply:
I like the way the tests would read, looking from a 30k feet view because it would allow us to be more explicit on what each test is really testing.
Maybe this exercise would display even other places where we are doing redundant test assertions.
Nevertheless, I think we should do this in a separate PR.